pkgs/development/python-modules/filedate/default.nix

```diff
@@ -0,0 +1,58 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pip,
+  python-dateutil,
+}:
+buildPythonPackage rec {
+  pname = "filedate";
+  version = "3.0";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "kubinka0505";
+    repo = pname;
+    rev = version;
+    hash = "sha256-HvuGP+QlUlfAUfFmaVVvtPHGdrbWVxghQipnqTTvAQc=";
+  };
+
+  unpackPhase = ''
+    cp -r $src/Files/* .
+    cp -r $src/ReadMe.md .
+  '';
+
+  # The repo stores everything in "src" and uses setup.py to move "src" ->
+  # "filedate" before calling setup() and then tries to rename "filedate" back
+  # to "src" after.
+  postPatch = ''
+    mv src filedate
+    substituteInPlace setup.py \
+      --replace-fail "__title__ = os.path.basename(os.path.dirname(os.path.dirname(__file__)))" '__title__ = "filedate"'
+    substituteInPlace setup.py \
+      --replace-fail "cleanup = True" "cleanup = False"
+
+    # Read the ReadMe.md in our patched location
+    substituteInPlace setup.py \
+      --replace-fail 'desc = open("../ReadMe.md", encoding = "UTF-8").read().rstrip()' 'desc = open("ReadMe.md", encoding = "UTF-8").read().rstrip()'
+
+    # Disable renaming "filedate" dir back to "src"
+    substituteInPlace setup.py \
+      --replace-fail "if os.path.exists(__title__):" ""
+    substituteInPlace setup.py \
+      --replace-fail "	os.rename(__title__, directory)" ""
+  '';
```

Comment:
How about using `sourceRoot = "source/Files";`?

Reply:
My only question is how do I then grab the README.md file that is located at the root?

Reply:
I don't think we need README.md in our package.

Reply:
The [setup.py](https://github.com/kubinka0505/filedate/blob/master/Files/setup.py#L33-L38) reads the README.md for the description. I don't mind updating my PR so that the setup.py doesn't try to read the README.md, but, just confirming that's the intention.

Reply:
Ah, yes, please try `sourceRoot`.
The readme should already be in the right place.

Reply:
Sweeeeet. Got this working now. Thank you 🙏